packages/lib-utils/src/components/status/StatusBox.tsx

```diff
@@ -0,0 +1,99 @@
+import { Alert, EmptyState, EmptyStateBody, Spinner, Text, Title } from '@patternfly/react-core';
+import * as _ from 'lodash-es';
+import * as React from 'react';
+
+export type LoadError = {
+  message: string;
+  status: number;
+};
+
+type AccessDeniedProps = {
+  message: string | React.ComponentType;
+};
+
+export const AccessDenied: React.FC<AccessDeniedProps> = ({
+  message = 'Insufficient access rights.',
+}) => (
+  <EmptyState>
+    <Title headingLevel="h2">Restricted Access</Title>
+    <EmptyStateBody>
+      <Text>You don&apos;t have access to this section due to cluster policy.</Text>
+      {message && (
+        <Alert isInline variant="danger" title="Error details">
+          {message}
+        </Alert>
+      )}
+    </EmptyStateBody>
+  </EmptyState>
+);
+
+type StatusBoxProps = {
+  areFiltersApplied?: boolean;
+  data?: unknown;
```

Comment:
@fhlavac FWIW `unknown` should be avoided. Can we use a shape?

Reply:
@florkbr I will probably use `Record<string, unknown>`. Is that ok?

Reply:
@fhlavac @florkbr I like to make a callback to my point around `unknown` I [made on slack](https://coreos.slack.com/archives/C02EQERSV3N/p1646162496757559). I still believe this is true... types need to be Generic long before we consider `unknown` valid. `unknown` is a step up from `any` and as an SDK we will definitely have `unknown`s... but we need to make sure this is not something that can be properly typed. Otherwise the code that receives `data` will be effectively `any`.

Reply:
@andrewballantyne @florkbr added a generic type to `StatusBoxProps`. In this case, the `data` property is used just to verify if it is empty or not so it's probably enough to expect that it is an array.

Reply:
@andrewballantyne agreed; I pinned your comment in the forum as well so we can find it again :)

Reply:
@fhlavac for now that's ok. One thing of note - if we're only checking `data` to see if it's empty, we could just add a new prop `empty: boolean;` and do the check in the consumer (that way we don't need to know anything about the data structure here).